avoid bc 1.07 or later, possibly hanging in the future!
see: https://bugs.archlinux.org/task/53546


this new diff is for kernel 5.4 stable branch
diff --git a/Kbuild b/Kbuild
index 3109ac786e76..be69d7f81def 100644
--- a/Kbuild
+++ b/Kbuild
@@ -18,7 +18,7 @@ $(bounds-file): kernel/bounds.s FORCE
 
 timeconst-file := include/generated/timeconst.h
 
-filechk_gentimeconst = echo $(CONFIG_HZ) | bc -q $<
+filechk_gentimeconst = echo "$(CONFIG_HZ);" | bc -q $<
 
 $(timeconst-file): kernel/time/timeconst.bc FORCE
 	$(call filechk,gentimeconst)
#---------
#for kernel 5.0-rc1, 5.3.13 patch:
#diff --git a/Kbuild b/Kbuild
#index 65db5bef2e36..c00d64b3eb85 100644
#--- a/Kbuild
#+++ b/Kbuild
#@@ -26,7 +26,7 @@ timeconst-file := include/generated/timeconst.h
# 
# targets += $(timeconst-file)
# 
#-filechk_gentimeconst = echo $(CONFIG_HZ) | bc -q $<
#+filechk_gentimeconst = echo "$(CONFIG_HZ);" | bc -q $<
# 
# $(timeconst-file): kernel/time/timeconst.bc FORCE
# 	$(call filechk,gentimeconst)
